### PR TITLE
Comply with guide for method doc: array.c

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Methods considered:
- count
- flatten!
- flatten
- cycle
- permutation
- combination
- repeated_permutation
- repeated_combination
- product
- take
- take_while
- drop
- drop_while

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
@jeremyevans, is there a bug report associated with the rdoc link problem?

**Comment by jeremyevans** [comment]
> @jeremyevans, is there a bug report associated with the rdoc link problem?

https://github.com/ruby/docs.ruby-lang.org/pull/105

**Comment by jeremyevans** [comment]
> > @jeremyevans, is there a bug report associated with the rdoc link problem?
> 
> [ruby/docs.ruby-lang.org#105](https://github.com/ruby/docs.ruby-lang.org/pull/105)
> 
> This is a PR, yes? I meant a defect report.

It's not a bug in Ruby itself, so it wouldn't be filed as a Redmine bug.  You could file an issue in the docs.ruby-lang.org repository, but that seems pointless as the pull request already exists. I suppose you could add a Misc issue in Redmine if you would like to bring more attention to the issue.

**Comment by jeremyevans** [comment]
Also, you may not be aware, but you edited my comment instead of replying to it, which makes it look like I'm asking a question of myself.  It's probably best not to edit comments of other users unless they are saying something that violates our code of conduct.

**Comment by BurdetteLamar** [comment]
> 
> 
> Also, you may not be aware, but you edited my comment instead of replying to it, which makes it look like I'm asking a question of myself. It's probably best not to edit comments of other users unless they are saying something that violates our code of conduct.

Sorry, I thought you'd edited mine.  Until I thought that,, I never knew I _could_ edit not-mine.

**Comment by BurdetteLamar** [comment]
> 
> 
> > > @jeremyevans, is there a bug report associated with the rdoc link problem?
> > 
> > 
> > [ruby/docs.ruby-lang.org#105](https://github.com/ruby/docs.ruby-lang.org/pull/105)
> > This is a PR, yes? I meant a defect report.
> 
> It's not a bug in Ruby itself, so it wouldn't be filed as a Redmine bug. You could file an issue in the docs.ruby-lang.org repository, but that seems pointless as the pull request already exists. I suppose you could add a Misc issue in Redmine if you would like to bring more attention to the issue.

Thanks.  This is exactly what I wanted to know.

File: array.c [hunk 1]
```diff
@@ -6144,8 +6134,6 @@ rb_ary_flatten_bang(int argc, VALUE *argv, VALUE ary)
  *  - Each non-Array element is unchanged.
  *  - Each \Array is replaced by its individual elements.
  *
- *  Argument +level+, if given and not +nil+, must be Integers.
- *
  *  With non-negative argument +level+, flattens recursively through +level+ levels:
```
Comment (jeremyevans) [inline array.c:6137]: Not a big deal, but you may want to use `non-negative \Integer argument` as you did above.

File: array.c [hunk 2]
```diff
@@ -6379,11 +6359,7 @@ rb_ary_cycle_size(VALUE self, VALUE args, VALUE eobj)
  *    array.cycle -> new_enumerator
  *    array.cycle(count) -> new_enumerator
  *
- *  Argument +count+, if given, must be +nil+ or an \Integer.
- *
- *  ---
- *
- *  When called with positive argument +count+ and a block,
+ *  When called with positive \Intgeger argument +count+ and a block,
```
Comment (jeremyevans) [inline array.c:6362]: Typo here, should be `\Integer`.

File: array.c (hunk 1 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c (hunk 2 above)
Comment (BurdetteLamar) [reply array.c]: Fixed.

**Pull request merged**